diff --git a/Q01_footfall_counter/src/main.cpp b/Q01_footfall_counter/src/main.cpp
index b27937d..0b580dd 100644
--- a/Q01_footfall_counter/src/main.cpp
+++ b/Q01_footfall_counter/src/main.cpp
@@ -55,6 +55,8 @@
 
 #include "draw.h"
 
+#define DISABLE_AI
+
 /*****************************************
 * Global Variables
 ******************************************/
@@ -1228,23 +1230,26 @@ void mipi_cam_init(void)
 {
     int ret = 0;
     std::cout << "[INFO] MIPI CAM Init \n";
-    const char *commands[4] =
+
+    constexpr size_t NUM_COMMANDS = 5U;
+    std::array<std::string, NUM_COMMANDS> const commands =
     {
-        "media-ctl -d /dev/media0 -r",
-        "media-ctl -d /dev/media0 -V \"\'ov5645 0-003c\':0 [fmt:UYVY8_2X8/640x480 field:none]\"",
-        "media-ctl -d /dev/media0 -l \"\'rzg2l_csi2 10830400.csi2\':1 -> \'CRU output\':0 [1]\"",
-        "media-ctl -d /dev/media0 -V \"\'rzg2l_csi2 10830400.csi2\':1 [fmt:UYVY8_2X8/640x480 field:none]\""
+      "media-ctl -d /dev/media0 -r",
+      "media-ctl -d /dev/media0 -l \"'csi-10830400.csi2':1 -> 'cru-ip-10830000.video':0 [1]\"",
+      "media-ctl -d /dev/media0 -V \"'csi-10830400.csi2':1 [fmt:UYVY8_2X8/640x480 field:none]\"",
+      "media-ctl -d /dev/media0 -V \"'ov5645 0-003c':0 [fmt:UYVY8_2X8/640x480 field:none]\"",
+      "media-ctl -d /dev/media0 -V \"'cru-ip-10830000.video':0 [fmt:UYVY8_2X8/640x480 field:none]\""
     };
 
     /* media-ctl command */
-    for (int i = 0; i < 4; i++)
+    for (auto const& command : commands)
     {
-        std::cout << commands[i] << "\n";
-        ret = system(commands[i]);
+        std::cout << command << "\n";
+        ret = system(command.c_str());
         std::cout << "system ret = " << ret << "\n";
         if (ret < 0)
         {
-            std::cout << "[ERROR]" << __func__ << ": failed media-ctl commands. index = " << i << "\n";
+            std::cout << "[ERROR]" << __func__ << ": failed media-ctl command:" << command << "\n";
             return;
         }
     }
@@ -1327,9 +1332,11 @@ int32_t main(int32_t argc, char * argv[])
 
     /*Disable OpenCV Accelerator due to the use of multithreading */
     #ifdef V2H
+#ifndef DISABLE_AI
     unsigned long OCA_list[16];
     for(int i = 0; i < 16; i++) OCA_list[i] = 0;
     OCA_Activate(&OCA_list[0]);
+#endif
     #endif
 
     /*Multithreading Variables*/
@@ -1342,7 +1349,7 @@ int32_t main(int32_t argc, char * argv[])
     InOutDataType input_data_type;
     bool runtime_status = false;
     std::string gstreamer_pipeline;
-    
+
     if (argc < 2) 
     {
         std::cout << "[ERROR] Please specify Input Source\n";
@@ -1373,8 +1380,7 @@ int32_t main(int32_t argc, char * argv[])
             std::cout << "[INFO] MIPI CAMERA \n";
             mipi_cam_init();
             std::string media_port = query_device_status("CRU");
-            gstreamer_pipeline = "v4l2src device=" + media_port + " ! videoconvert ! appsink";
-
+            gstreamer_pipeline = "v4l2src device=" + media_port + " ! video/x-raw,format=UYVY,width=640,height=480,framerate=15/1 ! videoconvert ! appsink";
         }
         break;
         default:
@@ -1434,6 +1440,7 @@ int32_t main(int32_t argc, char * argv[])
     }
     data_read();
 
+#ifndef DISABLE_AI
     int drpai_fd = open("/dev/drpai0", O_RDWR);
     if (0 > drpai_fd)
     {
@@ -1443,6 +1450,7 @@ int32_t main(int32_t argc, char * argv[])
     }
     /* Set drpai mem start address */
     uint64_t drpaimem_addr_start = 0;
+#endif
 
     /*Load Label from label_list file*/
     label_file_map = load_label_file(label_list);
@@ -1453,6 +1461,7 @@ int32_t main(int32_t argc, char * argv[])
         goto end_main;
     }
 
+#ifndef DISABLE_AI
     /*Load model_dir structure and its weight to runtime object */
     drpaimem_addr_start = init_drpai(drpai_fd);
    
@@ -1487,6 +1496,7 @@ int32_t main(int32_t argc, char * argv[])
         fprintf(stderr, "[ERROR] Input data type : neither FP32 nor FP16.\n");
         goto end_main;
     }
+#endif
 
     /*Termination Request Semaphore Initialization*/
     /*Initialized value at 1.*/
@@ -1498,6 +1508,7 @@ int32_t main(int32_t argc, char * argv[])
         goto end_threads;
     }
 
+#ifndef DISABLE_AI
     /*Create Inference Thread*/
     create_thread_ai = pthread_create(&ai_inf_thread, NULL, R_Inf_Thread, NULL);
     if (0 != create_thread_ai)
@@ -1507,6 +1518,7 @@ int32_t main(int32_t argc, char * argv[])
         ret_main = -1;
         goto end_threads;
     }
+#endif
 
     /* Create Capture Thread */
     create_thread_capture = pthread_create(&capture_thread, NULL, R_Capture_Thread, (void *) &gstreamer_pipeline);
@@ -1546,6 +1558,7 @@ end_threads:
             ret_main = -1;
         }
     }
+#ifndef DISABLE_AI
     if (0 == create_thread_ai)
     {
         ret = wait_join(&ai_inf_thread, AI_THREAD_TIMEOUT);
@@ -1555,6 +1568,7 @@ end_threads:
             ret_main = -1;
         }
     }
+#endif
     if (0 == create_thread_key)
     {
         ret = wait_join(&kbhit_thread, EXIT_THREAD_TIMEOUT);
@@ -1570,6 +1584,7 @@ end_threads:
     {
         sem_destroy(&terminate_req_sem);
     }
+#ifndef DISABLE_AI
     goto end_close_drpai;
 end_close_drpai:
     /*Close DRP-AI Driver.*/
@@ -1583,6 +1598,7 @@ end_close_drpai:
             ret_main = -1;
         }
     }
+#endif
     goto end_main;
 
 end_main:
