diff --git a/R01_object_detection/src/main.cpp b/R01_object_detection/src/main.cpp
index 3d309c6..abaf124 100755
--- a/R01_object_detection/src/main.cpp
+++ b/R01_object_detection/src/main.cpp
@@ -66,6 +66,8 @@
 /*Mutual exclusion*/
 #include <mutex>
 
+#define DISABLE_AI
+
 /*****************************************
 * Global Variables
 ******************************************/
@@ -650,24 +652,28 @@ int8_t print_result(Image* img, cv::Mat& proc_image)
 #ifndef V2H /* V2L */
 void mipi_cam_init(void)
 {
-    int8_t ret = 0;
-    const char* commands[4] =
+    int ret = 0;
+    std::cout << "[INFO] MIPI CAM Init \n";
+
+    constexpr size_t NUM_COMMANDS = 5U;
+    std::array<std::string, NUM_COMMANDS> const commands =
     {
-        "media-ctl -d /dev/media0 -r",
-        "media-ctl -d /dev/media0 -V \"\'ov5645 0-003c\':0 [fmt:UYVY8_2X8/640x480 field:none]\"",
-        "media-ctl -d /dev/media0 -l \"\'rzg2l_csi2 10830400.csi2\':1 -> \'CRU output\':0 [1]\"",
-        "media-ctl -d /dev/media0 -V \"\'rzg2l_csi2 10830400.csi2\':1 [fmt:UYVY8_2X8/640x480 field:none]\""
+      "media-ctl -d /dev/media0 -r",
+      "media-ctl -d /dev/media0 -l \"'csi-10830400.csi2':1 -> 'cru-ip-10830000.video':0 [1]\"",
+      "media-ctl -d /dev/media0 -V \"'csi-10830400.csi2':1 [fmt:UYVY8_2X8/640x480 field:none]\"",
+      "media-ctl -d /dev/media0 -V \"'ov5645 0-003c':0 [fmt:UYVY8_2X8/640x480 field:none]\"",
+      "media-ctl -d /dev/media0 -V \"'cru-ip-10830000.video':0 [fmt:UYVY8_2X8/640x480 field:none]\""
     };
 
     /* media-ctl command */
-    for (int i=0; i<4; i++)
+    for (auto const& command : commands)
     {
-        printf("%s\n", commands[i]);
-        ret = system(commands[i]);
-        printf("system ret = %d\n", ret);
-        if (ret<0)
+        std::cout << command << "\n";
+        ret = system(command.c_str());
+        std::cout << "system ret = " << ret << "\n";
+        if (ret < 0)
         {
-            printf("%s: failed media-ctl commands. index = %d\n", __func__, i);
+            std::cout << "[ERROR]" << __func__ << ": failed media-ctl command:" << command << "\n";
             return;
         }
     }
@@ -932,10 +938,11 @@ void *R_Capture_Thread(void *threadid)
                     mtx.lock();
                     /*Image: CAM_IMAGE_WIDTH*CAM_IMAGE_HEIGHT (BGR) */
                     input_image = g_frame.clone();
-                    
+#ifndef DISABLE_AI
                     /*Add padding for keeping the aspect ratio: CAM_IMAGE_WIDTH*CAM_IMAGE_WIDTH (BGR) */
                     cv::vconcat(input_image, padding_frame, input_image);
-                    
+#endif
+
                     /*Copy input data to drpai_buf for DRP-AI Pre-processing Runtime.*/
                     memcpy( drpai_buf->mem, input_image.data, drpai_buf->size);
                     /* Flush buffer */
@@ -944,6 +951,7 @@ void *R_Capture_Thread(void *threadid)
                     {
                         goto err;
                     }
+
                     mtx.unlock();
                     inference_start.store(1); /* Flag for AI Inference Thread. */
                 }
@@ -1357,6 +1365,7 @@ int32_t main(int32_t argc, char * argv[])
     }
 
     #ifdef V2H
+#ifndef DISABLE_AI
     /*Disable OpenCV Accelerator due to the use of multithreading */
     #ifdef V2N
     unsigned long OCA_list[16];
@@ -1366,6 +1375,7 @@ int32_t main(int32_t argc, char * argv[])
     for (int i=0; i < OCA_LIST_NUM; i++) OCA_list[i] = 0;
     #endif
     OCA_Activate( &OCA_list[0] );
+#endif
     printf("Disable OpenCVA\n");
     
     std::string media_port = query_device_status("usb");
@@ -1374,8 +1384,11 @@ int32_t main(int32_t argc, char * argv[])
     /* MIPI Camera Setup */
     mipi_cam_init();
     #endif
-    gstreamer_pipeline = "v4l2src device=" + media_port +" io-mode=dmabuf ! video/x-raw, width="+std::to_string(CAM_IMAGE_WIDTH)+", height="+std::to_string(CAM_IMAGE_HEIGHT)+" ,framerate=30/1 ! videoconvert ! appsink -v";
-
+#ifndef DISABLE_AI
+    gstreamer_pipeline = "v4l2src device=" + media_port + " io-mode=dmabuf ! video/x-raw,format=UYVY,width="+std::to_string(CAM_IMAGE_WIDTH)+", height="+std::to_string(CAM_IMAGE_HEIGHT)+" ,framerate=30/1 ! videoconvert ! appsink -v";
+#else
+    gstreamer_pipeline = "v4l2src device=" + media_port + " io-mode=dmabuf ! videoconvert ! appsink -v";
+#endif
     printf("RZ/V AI SDK Sample Application\n");
     printf("Model : Darknet YOLOv3 | %s\n", model_dir.c_str());
     printf("Input : %s\n", INPUT_CAM_NAME);
@@ -1411,6 +1424,7 @@ int32_t main(int32_t argc, char * argv[])
 
     /*DRP-AI Driver initialization*/
     errno = 0;
+
     drpai_fd = open("/dev/drpai0", O_RDWR);
     if (0 > drpai_fd)
     {
@@ -1503,6 +1517,7 @@ int32_t main(int32_t argc, char * argv[])
         ret_main = -1;
         goto end_threads;
     }
+
     /*Create Inference Thread*/
     create_thread_ai = pthread_create(&ai_inf_thread, NULL, R_Inf_Thread, NULL);
     if (0 != create_thread_ai)
@@ -1512,6 +1527,7 @@ int32_t main(int32_t argc, char * argv[])
         ret_main = -1;
         goto end_threads;
     }
+
     /*Create Capture Thread*/
     create_thread_capture = pthread_create(&capture_thread, NULL, R_Capture_Thread, NULL);
     if (0 != create_thread_capture)
@@ -1586,6 +1602,7 @@ end_threads:
             ret_main = -1;
         }
     }
+
     if (0 == create_thread_ai)
     {
         ret = wait_join(&ai_inf_thread, AI_THREAD_TIMEOUT);
@@ -1595,6 +1612,7 @@ end_threads:
             ret_main = -1;
         }
     }
+
     if (0 == create_thread_key)
     {
         ret = wait_join(&kbhit_thread, KEY_THREAD_TIMEOUT);
@@ -1616,16 +1634,21 @@ end_threads:
     goto end_close_dmabuf;
 
 end_close_dmabuf:
+
     buffer_free_dmabuf(drpai_buf);
+
     goto end_free_malloc;
 
 end_free_malloc:
+
     free(drpai_buf);
+
     drpai_buf = NULL;
     
     goto end_close_drpai;
 
 end_close_drpai:
+
     /*Close DRP-AI Driver.*/
     if (0 < drpai_fd)
     {
@@ -1637,6 +1660,7 @@ end_close_drpai:
             ret_main = -1;
         }
     }
+
     goto end_main;
 
 end_main:
