From 736ffd26c32942e744f0e773b89481c0a557f045 Mon Sep 17 00:00:00 2001
From: Asal Mirzaieva <asa.mirzaieva@canonical.com>
Date: Fri, 20 Feb 2026 17:46:13 +0100
Subject: [PATCH] Q01: Make camera and inferencing work

---
 Q01_footfall_counter/src/main.cpp | 23 ++++++++++++++---------
 1 file changed, 14 insertions(+), 9 deletions(-)

diff --git a/Q01_footfall_counter/src/main.cpp b/Q01_footfall_counter/src/main.cpp
index b27937d..3d40031 100644
--- a/Q01_footfall_counter/src/main.cpp
+++ b/Q01_footfall_counter/src/main.cpp
@@ -1228,18 +1228,22 @@ void mipi_cam_init(void)
 {
     int ret = 0;
     std::cout << "[INFO] MIPI CAM Init \n";
-    const char *commands[4] =
+    std::string csi = "csi-10830400.csi2";
+    std::string ip = "cru-ip-10830000.video";
+    const int commands_size = 5;
+
+    const char *commands[commands_size] =
     {
-        "media-ctl -d /dev/media0 -r",
-        "media-ctl -d /dev/media0 -V \"\'ov5645 0-003c\':0 [fmt:UYVY8_2X8/640x480 field:none]\"",
-        "media-ctl -d /dev/media0 -l \"\'rzg2l_csi2 10830400.csi2\':1 -> \'CRU output\':0 [1]\"",
-        "media-ctl -d /dev/media0 -V \"\'rzg2l_csi2 10830400.csi2\':1 [fmt:UYVY8_2X8/640x480 field:none]\""
+      "media-ctl -d /dev/media0 -r",
+      "media-ctl -d /dev/media0 -l \"'csi-10830400.csi2':1 -> 'cru-ip-10830000.video':0 [1]\"",
+      "media-ctl -d /dev/media0 -V \"'csi-10830400.csi2':1 [fmt:UYVY8_2X8/640x480 field:none]\"",
+      "media-ctl -d /dev/media0 -V \"'ov5645 0-003c':0 [fmt:UYVY8_2X8/640x480 field:none]\"",
+      "media-ctl -d /dev/media0 -V \"'cru-ip-10830000.video':0 [fmt:UYVY8_2X8/640x480 field:none]\"",
     };
 
     /* media-ctl command */
-    for (int i = 0; i < 4; i++)
+    for (int i = 0; i < commands_size; i++)
     {
-        std::cout << commands[i] << "\n";
         ret = system(commands[i]);
         std::cout << "system ret = " << ret << "\n";
         if (ret < 0)
@@ -1372,8 +1376,9 @@ int32_t main(int32_t argc, char * argv[])
         {
             std::cout << "[INFO] MIPI CAMERA \n";
             mipi_cam_init();
-            std::string media_port = query_device_status("CRU");
-            gstreamer_pipeline = "v4l2src device=" + media_port + " ! videoconvert ! appsink";
+            //std::string media_port = query_device_status("CRU");
+            std::string media_port = "/dev/video0";
+            gstreamer_pipeline = "v4l2src device=" + media_port + " ! video/x-raw,format=UYVY,width=640,height=480,framerate=15/1 ! videoconvert ! appsink";
 
         }
         break;
-- 
2.43.0

