From 52c9f36b526b4743c46526723d6247e15e785df2 Mon Sep 17 00:00:00 2001
From: Asal Mirzaieva <asa.mirzaieva@canonical.com>
Date: Wed, 25 Feb 2026 09:20:22 +0100
Subject: [PATCH] R01: Fix camera config and gstreamer pipeline

---
 R01_object_detection/src/main.cpp | 20 ++++++++++++--------
 1 file changed, 12 insertions(+), 8 deletions(-)

diff --git a/R01_object_detection/src/main.cpp b/R01_object_detection/src/main.cpp
index 3d309c6..44988ae 100755
--- a/R01_object_detection/src/main.cpp
+++ b/R01_object_detection/src/main.cpp
@@ -651,16 +651,19 @@ int8_t print_result(Image* img, cv::Mat& proc_image)
 void mipi_cam_init(void)
 {
     int8_t ret = 0;
-    const char* commands[4] =
-    {
+    const int commands_size = 5;
+
+    const char *commands[commands_size] =
+      {
         "media-ctl -d /dev/media0 -r",
-        "media-ctl -d /dev/media0 -V \"\'ov5645 0-003c\':0 [fmt:UYVY8_2X8/640x480 field:none]\"",
-        "media-ctl -d /dev/media0 -l \"\'rzg2l_csi2 10830400.csi2\':1 -> \'CRU output\':0 [1]\"",
-        "media-ctl -d /dev/media0 -V \"\'rzg2l_csi2 10830400.csi2\':1 [fmt:UYVY8_2X8/640x480 field:none]\""
-    };
+        "media-ctl -d /dev/media0 -l \"'csi-10830400.csi2':1 -> 'cru-ip-10830000.video':0 [1]\"",
+        "media-ctl -d /dev/media0 -V \"'csi-10830400.csi2':1 [fmt:UYVY8_2X8/640x480 field:none]\"",
+        "media-ctl -d /dev/media0 -V \"'ov5645 0-003c':0 [fmt:UYVY8_2X8/640x480 field:none]\"",
+        "media-ctl -d /dev/media0 -V \"'cru-ip-10830000.video':0 [fmt:UYVY8_2X8/640x480 field:none]\"",
+      };
 
     /* media-ctl command */
-    for (int i=0; i<4; i++)
+    for (int i = 0; i < commands_size; i++)
     {
         printf("%s\n", commands[i]);
         ret = system(commands[i]);
@@ -1374,7 +1377,8 @@ int32_t main(int32_t argc, char * argv[])
     /* MIPI Camera Setup */
     mipi_cam_init();
     #endif
-    gstreamer_pipeline = "v4l2src device=" + media_port +" io-mode=dmabuf ! video/x-raw, width="+std::to_string(CAM_IMAGE_WIDTH)+", height="+std::to_string(CAM_IMAGE_HEIGHT)+" ,framerate=30/1 ! videoconvert ! appsink -v";
+    media_port = "/dev/video0";
+    gstreamer_pipeline = "v4l2src device=" + media_port +" io-mode=dmabuf ! video/x-raw,format=UYVY,width="+std::to_string(CAM_IMAGE_WIDTH)+", height="+std::to_string(CAM_IMAGE_HEIGHT)+" ,framerate=30/1 ! videoconvert ! appsink -v";
 
     printf("RZ/V AI SDK Sample Application\n");
     printf("Model : Darknet YOLOv3 | %s\n", model_dir.c_str());
-- 
2.43.0

