diff --git a/Q01_footfall_counter/src/main.cpp b/Q01_footfall_counter/src/main.cpp
index fa1b0b5..ac70c52 100644
--- a/Q01_footfall_counter/src/main.cpp
+++ b/Q01_footfall_counter/src/main.cpp
@@ -1375,7 +1375,7 @@ int32_t main(int32_t argc, char * argv[])
             std::cout << "[INFO] MIPI CAMERA \n";
             mipi_cam_init();
             std::string media_port = query_device_status("CRU");
-            gstreamer_pipeline = "v4l2src device=" + media_port + " ! video/x-raw,format=YUY2,width=640,height=480 ! videoconvert ! appsink";
+            gstreamer_pipeline = "v4l2src device=" + media_port + " ! video/x-raw,format=UYVY,width=640,height=480 ! videoconvert ! appsink";
 
         }
         break;
